target/linux/ath79/files/drivers/mtd/nand/rb4xx_nand.c

```diff
@@ -0,0 +1,399 @@
+/*
+ *  NAND flash driver for the MikroTik RouterBoard 4xx series
+ *
+ *  Copyright (C) 2008-2011 Gabor Juhos <juhosg@openwrt.org>
+ *  Copyright (C) 2008 Imre Kaloz <kaloz@openwrt.org>
+ *
+ *  This file was based on the driver for Linux 2.6.22 published by
+ *  MikroTik for their RouterBoard 4xx series devices.
+ *
+ *  This program is free software; you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License version 2 as published
+ *  by the Free Software Foundation.
+ */
+
+#include <linux/version.h>
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/init.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,14,0)
+#include <linux/mtd/nand.h>
+#else
+#include <linux/mtd/rawnand.h>
+#endif
+#include <linux/mtd/mtd.h>
+#include <linux/mtd/partitions.h>
+#include <linux/platform_device.h>
+#include <linux/delay.h>
+#include <linux/io.h>
+#include <linux/gpio.h>
+#include <linux/slab.h>
+#include <linux/of.h>
+
+#include <asm/mach-ath79/ath79.h>
+#include <asm/mach-ath79/rb4xx_cpld.h>
+
+#define DRV_NAME        "rb4xx-nand"
+#define DRV_VERSION     "0.2.1"
```

Comment:
Version bump 0.2.0 -> 0.2.1

Reply:
Note that this way of documenting changes is useless, as it will be lost as soon as you PR is merged. Please add similar comments (if they are important enough) to the commit message where they will be preserved.

Despite, you are adding so much driver code, I wonder whether it wouldn't be better to do that in separate commits (which would also make providing details about them easier).
Note that I cannot review this driver stuff, so my comments here are mostly organizational.

Reply:
I see. In that case I will break this up into 3 smaller commits. The first a simple file copy, the second the patches for the drivers, and then the third the rb493g parts.